### PR TITLE
Implement support for App Platform log_destinations

### PR DESCRIPTION
**Pull request by jbrunton**
Possible PR for #796.

I branched from #797 and this PR uses that as a base branch, so that I could build on the changes by @andrewsomething for alerts. Hence, that should get merged first.

This is my first time contributing to this repo so I'd appreciate a careful review.

### CODE REVIEW

File: digitalocean/app_spec.go [hunk 1]
```diff
@@ -700,6 +718,68 @@ func appSpecComponentAlerts() *schema.Resource {
...
+			"papertrail": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				MaxItems:    1,
+				Description: "Papertrail configuration.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"endpoint": {
+							Type:        schema.TypeString,
+							Optional:    true,
```
Comment (andrewsomething) [inline digitalocean/app_spec.go:738]:
```suggestion
							Required:    true,
```

We can set these to required as that will only apply if the parent is set. So If `papertrail` is used, `endpoint` is required.

File: digitalocean/app_spec.go [hunk 2]
```diff
@@ -700,6 +718,68 @@ func appSpecComponentAlerts() *schema.Resource {
...
+							Description: "Datadog HTTP log intake endpoint.",
+						},
+						"api_key": {
+							Type:        schema.TypeString,
+							Optional:    true,
```
Comment (andrewsomething) [inline digitalocean/app_spec.go:758]:
```suggestion
							Required:    true,
```

File: digitalocean/app_spec.go [hunk 3]
```diff
@@ -700,6 +718,68 @@ func appSpecComponentAlerts() *schema.Resource {
...
+			"logtail": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				MaxItems:    1,
+				Description: "Logtail configuration.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"token": {
+							Type:        schema.TypeString,
+							Optional:    true,
```
Comment (andrewsomething) [inline digitalocean/app_spec.go:773]:
```suggestion
							Required:    true,
```

**Review by andrewsomething — COMMENTED** [review]
Thanks for picking this up @jbrunton! Everything looks great to me. I did have a few tiny UX suggestions inline.

**Comment by andrewsomething** [comment]
https://github.com/digitalocean/terraform-provider-digitalocean/pull/797 was merged using "squash and merge." Unfortunately that makes the history here a bit messy. A rebase on masin (`git pull --rebase origin main`) should clean that up. Sorry for the noise.

**Comment by jbrunton** [comment]
@andrewsomething Thanks for the review! No worries, I was assuming a rebase might be required. I'll take a look at your comments this weekend.

**Comment by jbrunton** [comment]
@andrewsomething: Applied your suggestions and rebased.

**Comment by MattIPv4** [comment]
Hey, @jbrunton - Thanks a ton for this sweet PR! 😄

Would you please shoot me an email when you get a chance?
mcowley at digitalocean dot com 🎉

**Pull request merged**